jaeger-core/src/main/java/io/jaegertracing/Configuration.java

```diff
@@ -135,6 +137,11 @@
    */
   public static final String JAEGER_PROPAGATION = JAEGER_PREFIX + "PROPAGATION";
 
+  /**
+   * Whether to skip loading a metrics factory from the classpath. Boolean. Optional.
+   */
+  public static final String JAEGER_SKIP_METRICS_FACTORY_LOADER = JAEGER_PREFIX + "SKIP_METRICS_FACTORY_LOADER";
```

Comment:
Do we need an env var to skip? Wouldn't it be good enough to just not include the dependency on the metrics factory?

Reply:
We never know what sort of crazy use cases people have. For instance, there might be someone extending our Micrometer Metrics Factory to build their own factory, so, they would have both dependencies on the classpath.

Reply:
Isn't it better to use ENV to define impl which would you like to load?

Reply:
The most appropriate thing to do for a Java application is to try to load implementations from the classpath using the service loader. The env var/property here is to *skip* the service loader entirely. Requiring an env var to specify an implementation isn't really idiomatic and wouldn't be the right direction, IMO.

Reply:
Maybe I didn't express well myself... I would never require conf property to specify the impl...

With ServiceLoader I have seen patterns (e.g  jax-rs clientbuilder resolution) that it's possible to specify implementation by fully qualified class name in case if there are multiple impl on classpath. It's useful for test purposes or just  when you have multiple implementations on class path and you want to choose a specific one.

So In this case, if there are multiple impls on classpath I would expect a way how I can specify the impl I want... Or if the env/prop is defined let's load it if not pick the first(random!) one.

Reply:
I see now what you mean. 

How about we keep this idea as a separate issue and see whether this is actually needed? While the code for this would be easy to implement (`Class.forName()`, basically), it's more code, and it's a code we aren't really sure will be used.

Reply:
I see that as a common practice rather than skip parameter. I am +1 on removing skip parameter and if necessary add parameter with FQCN in a separate PR (if you don't want to handle it here)

Reply:
Alright, the latest commit just removed the skip parameter.